cogs/ebay.py

```diff
@@ -0,0 +1,218 @@
+        return embed
+
+    async def make_soup(self, searchTerm):
+        cleanSearchTerm = urllib.parse.quote(searchTerm) # clean up the search term for the url
+        url = f"https://www.ebay.co.uk/sch/i.html?_from=R40&_nkw={searchTerm}&_sacat=0&rt=nc&LH_Sold=1&LH_Complete=1&_ipg=200"
+        async with aiohttp.ClientSession() as session:
+            data = await session.get(url, headers=headers)
+            pageText = await data.text()
+            await session.close()
+        return soup(pageText, 'html.parser')
+
+    async def get_filter(self, searchTerm):
+        """Filters out listings with certain words"""
+        # define filtered words
+        searchTerm = searchTerm.split(' ')
+        edgeCases = {'max': ['amd', 'intel'],
+                     'pc': ['case'],
+                     'cooler': ['aio', 'air', 'liquid']}
+        words = ['pro', 'plus', 'max', 'super', 'bundle', 'combo', 'faulty',
+                 'ti', 'xt', 'spare', 'spares', 'repair', 'repairs', 'cooler',
+                 'pc', 'damaged', 'broken', 'with', 'for', 'dell', 'hp',
+                 'gigabyte', 'acer', 'lenovo', 'asus', 'alienware', 'parts',
+                 'charger', 'dock', 'case', 'replicator', 'keyboard', 'mini',
+                 'lite', 'crack', 'cracked', '5g']
+        words.extend([i.strip('-').lower() for i in searchTerm if i.startswith('-')])
+        # remove duplicates
+        words = list(dict.fromkeys(words))
+        # collect words that were added to the filter
+        filteredWords = [i for i in searchTerm if i.startswith('-')]
+        # search term with boolean operators removed
+        filteredSearchTerm = [i for i in searchTerm if not i.startswith('-')]
+        # remove words in search from filter
+        for word in words:
+            if word in filteredSearchTerm:
+                words.remove(word)
+                if word in filteredWords:
+                    filteredWords.remove(word)
+        # remove words from filter if searching for certain things
+        for key in edgeCases.keys():
+            for word in edgeCases[key]:
+                if word in filteredSearchTerm and key in words:
+                    words.remove(key)
+        return ' '.join(filteredSearchTerm), words
+
+    async def get_product_info(self, product):
+        try:
+            d = {}
+            d['title'] = product.find('h3', {'class': 's-item__title'}).contents[-1]
+            d['url'] = product.find('a')['href']
+            d['image'] = product.find('img')['src']
+            d['price'] = await self.parse_price(product)
+            d['ended_date'] = await self.parse_date(product)
+            d['id'] = d['url'].split('/')[-1].split('?')[0]
+            return {'title': str(d['title']), 'url': str(d['url']), 'image': str(d['image']), 'price': d['price'], 'ended_date': d['ended_date'], 'id': str(d['id'])}
+        except:
+            return None
+
+    async def parse_price(self, product):
```

Comment:
you reckon you could comment this to make it easier to digest what's going on please?